### ISSUE TITLE
Ethylene Dupe Bug

### ISSUE THREAD

**Issue by NillerMedDild** [mekanism/Mekanism#5887]
### Issue description:
When Ethylene in gas form is inserted from a Gas Tank into a Gas Input from Advanced Generators (screenshot 1), 500 mB of Ethylene is drained, and the Gas Input receives 1000 mB of Liquid Ethylene. After that initial drain of 500 mB, the tank keeps supplying liquid ethylene, but none is drained. 

The Advanced Generator correctly drains it's Gas Input buffer when the Gas Tank is removed.

This is effectively free power gen for 500mB of Ethylene.

#### Steps to reproduce:
1. Get a Basic Gas Tank of Ethylene
2. Build an Advanced Generator 
    Can be done with the following blocks, as a freeform multiblock: 
    * 1 Gas Input
    * 1 Gold Turbine
    * 1 Gas Turbine Controller
    * 1 Advanced Power Capacitor
3. Place the Basic Gas Tank next to the Gas Input, and set it to Extract to that side. 

#### Version (make sure you are on the latest version before reporting):
**Forge:** 1.12.2-14.23.5.2847

Reproduced with the following mods only:
- Mekanism-1.12.2-9.8.3.390
- MekanismGenerators-1.12.2-9.8.3.390
- generators-0.9.20.12-mc1.12.2
- bdlib-1.14.3.12-mc1.12.2
- jei_1.12.2-4.15.0.293

#### If a (crash)log is relevant for this issue, link it here: (_It's almost always relevant_)
[Latest.log in gist](https://gist.github.com/NillerMedDild/3e5e7e9cc5b4b2b9c4b71c5af769b3d2)

##### Gas Tank next to Gas Input
![billede](https://user-images.githubusercontent.com/21295394/74870880-c70e8100-535a-11ea-9118-72cb30852018.png)

##### Liquid Ethylene in Gas Input
![billede](https://user-images.githubusercontent.com/21295394/74871299-8b27eb80-535b-11ea-9a27-b24cb72ebc6c.png)

**Comment by pupnewfster** [comment]
We aren't actively developing 1.12 anymore, but I looked briefly to see if I could see a bug in their code, and I believe I found it. It seems they are implementing handling for our gas incorrectly. In [this line](https://github.com/bdew/generators/blob/mc1.12/src/net/bdew/generators/modules/gasInput/TileGasInput.scala#L34) they are always inserting the gas as a fluid into their machines if they can. This is a mistake, as they are entirely ignoring our `doTransfer` parameter which is used for simulation.

So in essence what they are doing, when we ask their tile to simulate inserting gas (so we can see how much would fit and calculation an even distribution), they are just ignoring the fact it is a simulation, and increasing their contents regardless.
